lib/db/schema/patch-045-046.sql

```diff
@@ -0,0 +1,188 @@
+    END IF;
+
+    INSERT INTO emails(
+        normalizedEmail,
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    )
+    VALUES(
+        LOWER(normalizedEmail),
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    );
+
+    COMMIT;
+END;
+
+CREATE PROCEDURE `accountEmails_1` (
+    IN `inUid` BINARY(16)
+)
+BEGIN
+    -- Return all emails that belong to the uid.
+    -- Email from the accounts table is considered the primary email.
+	(SELECT
+        a.normalizedEmail,
+        a.email,
+        a.uid,
+        a.emailCode,
+        a.emailVerified AS isVerified,
+        TRUE AS isPrimary,
+        a.createdAt AS verifiedAt,
+        a.createdAt
+    FROM
+        accounts a
+    WHERE
+        uid = LOWER(inUid))
+
+    UNION ALL
+
+    (SELECT
+        e.normalizedEmail,
+        e.email,
+        e.uid,
+        e.emailCode,
+        e.isVerified,
+        e.isPrimary,
+        e.verifiedAt,
+        e.createdAt
+    FROM
+        emails e
+    WHERE
+        uid = LOWER(inUid));
+END;
+
+CREATE PROCEDURE `deleteEmail_1` (
+    IN `inNormalizedEmail` VARCHAR(255)
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+        BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    -- Don't delete any email that are considered primary emails
+	SET @isPrimary = 0;
+    SELECT COUNT(*) INTO @isPrimary FROM accounts WHERE normalizedEmail = inNormalizedEmail;
+    IF @isPrimary > 0 THEN
+      SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = 'Can not delete a primary email address.';
+    END IF;
+
+	SELECT COUNT(*) INTO @isPrimary FROM emails WHERE normalizedEmail = inNormalizedEmail AND isPrimary = 1;
+    IF @isPrimary > 0 THEN
+      SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = 'Can not delete a primary email address.';
+    END IF;
+
+	DELETE FROM emails WHERE normalizedEmail = inNormalizedEmail;
+
+    COMMIT;
+END;
+
+DROP procedure IF EXISTS `deleteAccount_12`;
+
+CREATE PROCEDURE `deleteAccount_12`(
+  IN `uidArg` BINARY(16)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM sessionTokens WHERE uid = uidArg;
+  DELETE FROM keyFetchTokens WHERE uid = uidArg;
+  DELETE FROM accountResetTokens WHERE uid = uidArg;
+  DELETE FROM passwordChangeTokens WHERE uid = uidArg;
+  DELETE FROM passwordForgotTokens WHERE uid = uidArg;
+  DELETE FROM accounts WHERE uid = uidArg;
+  DELETE FROM devices WHERE uid = uidArg;
+  DELETE FROM unverifiedTokens WHERE uid = uidArg;
+  DELETE FROM unblockCodes WHERE uid = uidArg;
+  DELETE FROM emails WHERE uid = uidArg;
+
+  COMMIT;
+END;
+
+
+CREATE PROCEDURE `verifyEmail_4`(
+    IN `inUid` BINARY(16),
+    IN `inEmailCode` BINARY(16)
+)
+BEGIN
+    SET @updatedCount = 0;
+
+    IF (inEmailCode IS NULL) THEN
```

Comment:
Am I being overly cautious and accounting for `emailCode` not being present? I wanted to keep the previous functionality that just verified the the users account email, but I am not sure if that is actually needed.